shipstation_integration/orders.py

```diff
@@ -285,23 +292,57 @@ def create_erpnext_order(
 
 	so.save()
 
-	before_submit_hook = frappe.get_hooks("update_shipstation_order_before_submit")
-	if before_submit_hook:
-		so = frappe.get_attr(before_submit_hook[0])(store, so, order)
-		if so:
-			so.save()
-	if so:
-		so.submit()
-		frappe.db.commit()
-
-	after_submit_hook = frappe.get_hooks("update_shipstation_order_after_submit")
-	if before_submit_hook:
-		frappe.get_attr(after_submit_hook[0])(store, so, order)
-		frappe.db.commit()
+	set_status(so, order, store)
 
 	return so.name if so else None
 
 
+def set_status(
+	so: Union["SalesOrder", dict], order: "ShipStationOrder", store: "ShipstationStore"
+) -> None:
+
+	status_mapping = {
+		"awaiting_payment": ("Draft", 0),
+		"awaiting_shipment": ("To Deliver", 1),
+		"shipped": ("Completed", 1),
+		"on_hold": ("On Hold", 1),
+		"cancelled": ("Cancelled", 2),
+		"pending_fulfillment": ("To Deliver and Bill", 1),
+	}
```

Comment:
@batonac How is this status map being determined? Why is `awaiting_payment` equivalent to the Draft status? And should the `on_hold` status in Shipstation run the "On Hold" function against a Sales Order?

Reply:
Hey @Alchez, great questions~

My decision to mark order `awaiting_payment` as `Draft` was based on B2C conventions, where payment is required to confirm an order. This is the most common e-commerce scenario AFAIK. That said, I readily acknowledge this is not always the case. Perhaps this should be configured via a flag in `ShipStation Options`? We could add a "Mark Orders Awaiting Payment as Draft" boolean or "Awaiting Payment Status" select option to add configurability.

I'm honestly not familiar with a use-case for the On Hold status in Shipstation, but it feels like a sane decision to create a direct correlation between the two systems. If any of this becomes problematic, we could make all of these status correlations configurable.

Reply:
I await your direction on this one.

Reply:
@batonac The existence of a Sales Order does mean that payment has been remitted. In a marketplace context, settlements will not be on a per-order basis and will only happen periodically. It is not to correct to wait for the period close to further process an order. Awaiting Payment should align with "Submitted".

Reply:
That make sense.

Taking another stab at this. Here are the stock [ERPNext Sales Order statuses](https://docs.erpnext.com/docs/user/manual/en/sales-order#3-10-billing-and-delivery-status):
```
Draft
On Hold
To Deliver and Bill
To Bill
To Deliver
Completed
Cancelled
Closed
```
And the [ShipStation Order Statuses](https://help.shipstation.com/hc/en-us/articles/360025869712-Understanding-Order-Statuses):
```
Awaiting Payment
On Hold
Awaiting Shipment
Pending Fulfillment
Shipped
Cancelled
```
Here's the most apparent correlation to me:
| ShipStation Order Status | ERPNext Sales Order Status | ERPNext Sales Order Docstatus |
|--------------------------|----------------------------|-------------------------------|
| Awaiting Payment         | To Bill                    | Submitted                     |
| On Hold                  | On Hold                    | Submitted                     |
| Awaiting Shipment        | To Deliver                 | Submitted                     |
| Pending Fulfillment      | To Deliver                 | Submitted                     |
| Shipped                  | Completed                  | Submitted                     |
| Cancelled                | Cancelled                  | Cancelled                     |

'Pending Fulfillment` is the most obvious outlier here, which apparently [only applies to dropshipped orders](https://help.shipstation.com/hc/en-us/articles/360025869712-Understanding-Order-Statuses#:~:text=Orders%20you%20send,your%20ShipStation%20account.). Perhaps that should be moved to 'Completed' if there's no more that needs to be done by the ERP operator?

Reply:
I think this is correct, but we need the complete set for documentation's sake: 


| ShipStation Order Status | ERPNext Sales Order Status | ERPNext Sales Order Docstatus |
| ------------------------ | -------------------------- | ----------------------------- |
|                          | Draft                      | 1                             |
| Awaiting Payment         | To Bill                    | 1                             |
| On Hold                  | On Hold                    | 1                             |
| Awaiting Shipment        | To Deliver                 | 1                             |
| Pending Fulfillment      | To Deliver                 | 1                             |
| Shipped                  | Completed                  | 1                             |
| Cancelled                | Cancelled                  | 2                             |
|                          | Closed                     | 1                             |

"Closed" does not have an equivalent status in ShipStation, in that it means an order has been shorted by some amount. Draft also does not have an equivalent status in ShipStation, since the `ShipStationOrder` object is continuously mutable.

Reply:
The above looks right for a two-way integration, which I am interested in. For import only-scenario, can you check my latest commit? I simplified some things thanks to reverting to submit on import by default.